webrtc/protocol/sdes-dont-dont-dont.html

```diff
@@ -0,0 +1,48 @@
+<!doctype html>
+<meta charset=utf-8>
+<meta name="timeout" content="long">
+<title>RTCPeerConnection MUST NOT support SDES</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="../RTCPeerConnection-helper.js"></script>
+<script src="/webrtc/third_party/sdp/sdp.js"></script>
+<script>
+'use strict';
+
+// Test support for
+// https://www.rfc-editor.org/rfc/rfc8826#section-4.3.1
+
+const sdes_constraint = {'mandatory': {'DtlsSrtpKeyAgreement': false}};
+const sdp = `v=0
+o=- 0 3 IN IP4 127.0.0.1
+s=-
+t=0 0
+m=video 9 UDP/TLS/RTP/SAVPF 100
+c=IN IP4 0.0.0.0
+a=rtcp-mux
+a=sendonly
+a=mid:video
+a=rtpmap:100 VP8/90000
+a=fmtp:100 max-fr=30;max-fs=3600
+a=crypto:0 AES_CM_128_HMAC_SHA1_80 inline:2nra27hTUb9ilyn2rEkBEQN9WOFts26F/jvofasw
+a=ice-ufrag:ETEn
+a=ice-pwd:OtSK0WpNtpUjkY4+86js7Z/l
+`;
+
+promise_test(async t => {
+  const pc = new RTCPeerConnection(null, sdes_constraint);
+  t.add_cleanup(() => pc.close());
+
+  pc.addTransceiver('audio');
+  const offer = await pc.createOffer();
+  assert_false(offer.sdp.includes('\na=crypto:'));
+}, 'does not create offers with SDES');
+
+promise_test(async t => {
+  const pc = new RTCPeerConnection();
+  t.add_cleanup(() => pc.close());
+
+  pc.setRemoteDescription({type: 'offer', sdp})
+    .then(assert_unreached);
+}, 'rejects a remote offer that only includes SDES and no DTLS');
```

Comment:
@fippo The test itself isn't waiting for SRD to finish here, effectively testing that SRD does not resolve after `pc.close()`, which is correct behavior according to spec, but fails to test what this test says it's testing.

If I [let SRD finish](https://jsfiddle.net/jib1/6tbvwgu3/13/) (don't implement `add_cleanup`) then it fails in both Chrome an Firefox respectively with:

> Uncaught (in promise) DOMException: InvalidAccessError: Failed to execute 'setRemoteDescription' on 'RTCPeerConnection': Failed to set remote offer sdp: Called with SDP without DTLS fingerprint.

> Uncaught (in promise) DOMException: OperationError: Invalid description, no fingerprint attribute

Side-issue: It would be nice if the _"Uncaught (in promise) DOMException"_ messages in both browsers included the  `name` of the error, but they don't, so I [tested for that explicitly](https://jsfiddle.net/jib1/6tbvwgu3/19/) and have inserted those above.

We should update this test to check for failure. As to which browser has the correct error, that depends on whether we consider the SDP itself "valid" or not 🤷🏼‍♂️

<img width="755" alt="image" src="https://user-images.githubusercontent.com/3136226/132355163-1acceb65-81ba-494f-a181-0435b1662cda.png">

Reply:
https://github.com/w3c/webrtc-pc/issues/2672 -- different errors